### PR TITLE
[PROTOCOL-725] Liquidate Loan With NFT V2

### PR DESCRIPTION
**Pull request by admazzola**
When a loan using an NFT is liquidated, the NFT is sent to a Gnosis Safe.  This only worked for NFTv1 so this PR also adds this functionality to NFTv2

### CODE REVIEW

File: contracts/market/RepayFacet.sol [hunk 1]
```diff
@@ -42,7 +42,17 @@ import {
     LoanStatus
 } from "../storage/market.sol";
 
+import { TellerNFT_V2 } from "../nft/TellerNFT_V2.sol";
+
+import "hardhat/console.sol";
```
Comment (mkhoury93) [inline contracts/market/RepayFacet.sol:47]: please remove this

File: contracts/market/mainnet/MainnetRepayFacet.sol [hunk 2]
```diff
@@ -3,14 +3,36 @@ import { RepayFacet } from "../RepayFacet.sol";
...
+import { TellerNFT } from "../../nft/TellerNFT.sol";
+
 contract MainnetRepayFacet is RepayFacet {
+ 
+    TellerNFT private immutable TELLER_NFT_V1; 
+    
+    constructor(address nftV1Address, address nftV2Address)
+        RepayFacet(nftV2Address)
+    {
+        TELLER_NFT_V1 = TellerNFT(nftV1Address);
```
Comment (mkhoury93) [inline contracts/market/mainnet/MainnetRepayFacet.sol:15]: where are we using `TELLER_NFT_V1`

**Review by mkhoury93 — CHANGES_REQUESTED** [review]

File: contracts/market/RepayFacet.sol [hunk 3]
```diff
@@ -242,6 +250,25 @@ contract RepayFacet is RolesMods, ReentryMods, PausableMods, EscrowClaimTokens {
...
+
+        for (uint256 i; i < EnumerableSet.length(nfts); i++) {
+            uint256 nftId = EnumerableSet.at(nfts, i);
+
+            TELLER_NFT_V2.safeTransferFrom(
+                address(this),
+                AppStorageLib.store().nftLiquidationController,
+                nftId,
+                NFTLib.s().loanNFTsV2Amounts[loanID][nftId],
+                "0x"
```
Comment (admazzola) [inline contracts/market/RepayFacet.sol:267]: ""

File: contracts/market/mainnet/MainnetRepayFacet.sol [hunk 4]
```diff
@@ -13,4 +17,14 @@ contract MainnetRepayFacet is RepayFacet {
 
         super._restakeNFTForRepayment(loanID);
     }
+
+    /**
+       @notice On mainnet, we override the liquidateNFT method so that TellerNFTV1 is also transferred
```
Comment (admazzola) [inline contracts/market/mainnet/MainnetRepayFacet.sol:22]: asterisk

File: tasks/test.ts [hunk 5]
```diff
@@ -22,7 +22,7 @@ task('test').setAction(async (args, hre, runSuper) => {
   })
 
   // Disable logging
-  process.env.DISABLE_LOGS = 'true'
+  //  process.env.DISABLE_LOGS = 'true'
```
Comment (admazzola) [inline tasks/test.ts:25]: turn this back on

File: test/integration/liquidation.test.ts [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+
+              const totalOwedAfterRepay = await diamond.getTotalOwed(loanId)
+
+              const loanData = await diamond.getLoan(loanId)
+
+              borrowerBalance.should.eql(hre.toBN(200, lendingTokenDecimals))
+
+              totalOwedAfterRepay.should.eql(0)
+
+              expect(loanData.status).to.equal(LoanStatus.Closed)
```
Comment (admazzola) [inline test/integration/liquidation.test.ts:239]: test liquidate in v1

**Review by dwking2000 — DISMISSED** [review]
Looks good. Based on our walkthrough I have no comments.

File: contracts/market/RepayFacet.sol [hunk 7]
```diff
@@ -242,6 +250,25 @@ contract RepayFacet is RolesMods, ReentryMods, PausableMods, EscrowClaimTokens {
         NFTLib.restakeLinkedV2(loanID, LibLoans.loan(loanID).borrower);
     }
 
+    function _liquidateNFT(uint256 loanID) internal virtual {
+        //uint256 amt = NFTLib.s().stakedNFTsV2Amounts[owner][loanID];
+
```
Comment (passabilities) [inline contracts/market/RepayFacet.sol]: lets remove this

File: test/integration/loans.test.ts [hunk 8]
```diff
@@ -224,7 +224,7 @@ describe('Loans', () => {
               const totalOwedAfterRepay = await diamond.getTotalOwed(loanId)
 
               const loanData = await diamond.getLoan(loanId)
-
+
```
Comment (passabilities) [inline test/integration/loans.test.ts]: run `yarn format:write`

**Review by passabilities — CHANGES_REQUESTED** [review]

**Review by passabilities — APPROVED** [review]

**Review by mkhoury93 — APPROVED** [review]

**Review by dumebi — APPROVED** [review]

**Pull request merged**